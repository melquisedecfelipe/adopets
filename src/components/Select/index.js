import React from 'react';

import './styles.scss';

import { Select } from 'antd';

const { Option } = Select;

export default function SelectComponent({ onChange, option, placeholder, value }) {
  return (
    <Select value={value} onChange={onChange} placeholder={placeholder} showSearch>
      {option
        ? option.map(elem => (
            <Option value={elem.value} key={elem.value}>
              {elem.description}
            </Option>
          ))
        : null}
    </Select>
  );
}
