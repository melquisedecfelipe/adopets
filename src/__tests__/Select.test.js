import React from 'react';

import { shallow } from 'enzyme';

import Select from '../components/Select';

describe('Select testing with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Select />);

    expect(wrapper).toMatchSnapshot();
  });
});
