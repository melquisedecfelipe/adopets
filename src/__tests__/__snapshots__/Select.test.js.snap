// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select testing with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
